mlir/lib/Target/LLVMIR/Dialect/ROCDL/ROCDLToLLVMIRTranslation.cpp

```diff
@@ -134,7 +140,19 @@ class ROCDLDialectLLVMIRTranslationInterface
       llvmAttrValue.append(value.getValue());
       llvmFunc->addFnAttr("amdgpu-flat-work-group-size", llvmAttrValue);
     }
-
+    if (ROCDL::ROCDLDialect::getUniformWorkGroupSizeAttrName() ==
+        attribute.getName()) {
+      auto func = dyn_cast<LLVM::LLVMFuncOp>(op);
+      if (!func)
+        return failure();
```

Comment:
Can we add here an `emitError` indicating why it is failing?

Reply:
The surrounding code doesn't seem to `emitError` but instead to silently ignore these attributes when they're on non-function ops.

Do we want to change that convention uniformly?

Reply:
I definitely don't like the idea of failing and not knowing why, because if these where safe to ignore, then they should return `success`  instead of `failure`. So I think either we `emitError`s or if they are safe to ignore then we return `success` instead of `failure`.

Reply:
I'll go ahead and clean that up